core/src/main/java/org/mule/runtime/core/policy/DefaultPolicyManager.java

```diff
@@ -38,28 +42,55 @@
+                                                 Processor flowExecutionProcessor,
+                                                 MessageSourceResponseParametersProcessor messageSourceResponseParametersProcessor) {
+    PolicyPointcutParameters sourcePointcutParameters = createSourcePointcutParameters(sourceIdentifier, sourceEvent);
+    List<Policy> parameterizedPolicies = policyProvider.findSourceParameterizedPolicies(sourcePointcutParameters);
     if (parameterizedPolicies.isEmpty()) {
-      return empty();
+      return event -> {
+        try {
+          Event flowExecutionResult = flowExecutionProcessor.process(sourceEvent);
+          return right(new SuccessSourcePolicyResult(flowExecutionResult,
+                                                     messageSourceResponseParametersProcessor
+                                                         .getSuccessfulExecutionResponseParametersFunction()
+                                                         .apply(flowExecutionResult),
+                                                     messageSourceResponseParametersProcessor));
+        } catch (Exception e) {
+          MessagingException messagingException =
+              e instanceof MessagingException ? (MessagingException) e : new MessagingException(event, e, flowExecutionProcessor);
+          return Either.left(new FailureSourcePolicyResult(messagingException, messageSourceResponseParametersProcessor
+              .getFailedExecutionResponseParametersFunction()
+              .apply(messagingException.getEvent()), messageSourceResponseParametersProcessor));
+        }
+      };
     }
-    return of(new CompositeSourcePolicy(parameterizedPolicies, lookupSourceParametersTransformer(sourceIdentifier),
-                                        sourcePolicyFactory));
+    return new CompositeSourcePolicy(parameterizedPolicies,
+                                     lookupSourceParametersTransformer(sourceIdentifier),
+                                     sourcePolicyProcessorFactory, flowExecutionProcessor,
+                                     messageSourceResponseParametersProcessor);
   }
 
   @Override
-  public Optional<OperationPolicy> findOperationPolicy(String executionIdentifier, ComponentIdentifier operationIdentifier) {
-    List<Policy> parameterizedPolicies = policyProvider.findOperationParameterizedPolicies(operationIdentifier);
+  public OperationPolicy createOperationPolicy(ComponentIdentifier operationIdentifier, Event event,
+                                               Map<String, Object> operationParameters,
+                                               OperationExecutionFunction operationExecutionFunction) {
+
+    PolicyPointcutParameters operationPointcutParameters =
+        createOperationPointcutParameters(operationIdentifier, operationParameters);
+    List<Policy> parameterizedPolicies = policyProvider.findOperationParameterizedPolicies(operationPointcutParameters);
```

Comment:
I removed findOperationParameterizedPolicies from the API since it's only being used internally. And now it's private